Jenkinsfile

```diff
@@ -0,0 +1,168 @@
+                    expression { params.PLATFORM_FILTER == env.PLATFORM }
+                    expression { params.JDK_FILTER == 'all' }
+                    expression { params.JDK_FILTER == env.JDK_NAME }
+                } }
+                axes {
+                    axis {
+                        name 'JDK_NAME'
+                        values 'jdk_17_latest', 'jdk_21_latest'
+                    }
+                    axis {
+                        name 'PLATFORM'
+                        values 'ppc64le', 's390x', 'ubuntu'
+                    }
+                }
+                excludes {
+                    exclude {
+                        axis {
+                            name 'JDK_NAME'
+                            values 'jdk_21_latest'
+                        }
+                        axis {
+                            name 'PLATFORM'
+                            values 'ppc64le'
+                        }
+                    }
+                    exclude {
+                        axis {
+                            name 'JDK_NAME'
+                            values 'jdk_21_latest'
+                        }
+                        axis {
+                            name 'PLATFORM'
+                            values 's390x'
+                        }
+                    }
+                }
+                tools {
+                    jdk "${JDK_NAME}"
+                }
+                stages {
+                    stage('Clean workspace') {
+                        when {
+                            expression { params.CLEAN }
+                        }
+                        steps {
+                            sh 'git clean -fdx'
+                        }
+                    }
+
+                    stage('Build') {
+                        steps {
+                            echo "Do Build for ${PLATFORM}-${JDK_NAME}"
+                            sh 'java -version'
+                            sh "./mvnw -U $MAVEN_PARAMS -Dskip.camel.maven.plugin.tests -Darchetype.test.skip -Dmaven.test.skip.exec=true clean install"
+                        }
+                    }
+
+                    stage('Code Quality Review') {
+                        steps {
+                            script {
+                                if ("${PLATFORM}" == "ubuntu") {
+                                    withCredentials([string(credentialsId: 'apache-camel-core', variable: 'SONAR_TOKEN')]) {
+                                        /*
+                                        * sh "./mvnw $MAVEN_PARAMS -Dsonar.host.url=https://sonarcloud.io -Dsonar.java.experimental.batchModeSizeInKB=2048 -Dsonar.organization=apache -Dsonar.projectKey=apache_camel -Dsonar.branch.name=$BRANCH_NAME org.sonarsource.scanner.maven:sonar-maven-plugin:sonar"
+                                        */
+                                        echo "Code quality review ENABLED for ${PLATFORM}"
```

Comment:
This is a note for the community: we'll keep this one disabled until it is merged to avoid messing up with the sonar report. After the code is merged, I'll re-enable this.